File: authorization/src/main/kotlin/com/example/authorization/repository/UserRepository.java
package com.example.authorization.repository;

import com.example.authorization.entity.*;
import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.*;

import java.util.*;

@Repository
public interface UserRepository extends JpaRepository<User, Integer> {
    Optional<User> findUserByUsername(String username);

    Boolean existsByUsername(String username);
}

File: authorization/src/main/kotlin/com/example/authorization/repository/UserRepository.kt
package com.example.authorization.repository

import com.example.authorization.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Int> {
    fun findUserByUsername(username: String): User?
    fun existsByUsername(username: String): Boolean
    fun deleteByUsername(username: String)
}
